### ISSUE TITLE
Lua error blocks compilation

### ISSUE THREAD

**Issue by jliptrap** [Isti115/agda.nvim#6]
When fixing a compiler error, after loading I don't see any compiler output, instead just:
```
Error executing vim.schedule lua callback: ...trap/.config/nvim/packages/agda.nvim/lua/agda/output.lua:16: String cannot contain newlines
```
Invoking the compiler from the command-line at that point yields a "Failed to solve the following constraints" error.

Hopefully at some point I'll have a minimal repro to post here.  Meanwhile I may keep going back and forth between this plugin and [vim-agda](https://github.com/msuperdock/vim-agda) since the latter is less buggy though also less featureful.

**Comment by kands-code** [comment]
I also met this error, hope to fix

import:

```agda
import Relation.Binary.PropositionalEquality as Eq
open Eq using (_≡_; refl; cong; sym)
open Eq.≡-Reasoning using (begin_; _≡⟨⟩_; step-≡; _∎)
open import Data.Nat using (ℕ; zero; suc; _+_; _*_; _∸_)
```
what I want to do:
```agda
lemma₁ : ∀ (m : ℕ) → m * zero ≡ zero
lemma₁ zero = refl  
lemma₁ (suc m) rewrite lemma₁ m = refl

*-assoc : ∀ (m n p : ℕ) → (m * n) * p ≡ m * (n * p)
*-assoc m n zero rewrite lemma₁ (m * n) | lemma₁ n | lemma₁ m = {!   !}
*-assoc m n (suc p) = {!   !}
```
When I press `\r` in the first hole, I get this error

![图片](https://user-images.githubusercontent.com/59006329/149508802-4d89c126-bba7-4d80-a32b-63a5be44cfa4.png)

```
Error executing vim.schedule lua callback: .../share/nvim/sit
e/plugged/agda.nvim/lua/agda/response.lua:104: attempt to per
form arithmetic on local 'to_byte' (a nil value)
stack traceback:
        .../share/nvim/site/plugged/agda.nvim/lua/agda/respon
se.lua:104: in function 'cb'
        vim.lua:285: in function <vim.lua:285>
```

But it still worked and filled the hole with `refl`

This is really a confusing problem.

**Comment by isti115** [comment]
@kands-code Hi! Thanks for providing example code!
I was able to reproduce it and have provided a fix for the issue in c7da627547e978b4ac3780af1b8f418c8b12ff98.
Could you please verify that it works for you as well now without the error?

@jliptrap Is your problem still present?
I have iterated on the code responsible for output a few times since then, so I might have fixed it among those changes.

**Comment by kands-code** [comment]
> @kands-code Hi! Thanks for providing example code! I was able to reproduce it and have provided a fix for the issue in [c7da627](https://github.com/Isti115/agda.nvim/commit/c7da627547e978b4ac3780af1b8f418c8b12ff98). Could you please verify that it works for you as well now without the error?

Sorry for the late reply, this is now working, thanks!

**Comment by isti115** [comment]
Thanks for the confirmation, I will close the issue then. :slightly_smiling_face:
